src/main/java/com/yugabyte/sample/apps/SqlGeoPartitionedTable.java

```diff
@@ -0,0 +1,222 @@
+
+      if (tableOp.equals(TableOp.TruncateTable)) {
+        for (int i = 0; i < appConfig.numPartitions; i++) {
+          statement.execute(
+              String.format("TRUNCATE TABLE %s_%d", getTableName(), i));
+        }
+        LOG.info(String.format("Truncated table: %s", getTableName()));
+      }
+    }
+  }
+
+  public String getTableName() {
+    String tableName = appConfig.tableName != null ? appConfig.tableName : DEFAULT_TABLE_NAME;
+    return tableName.toLowerCase();
+  }
+
+  public String getRegion(Long key_number) {
+    return String.format("region_%d", key_number % appConfig.numPartitions);
+  }
+
+  private PreparedStatement getPreparedSelect() throws Exception {
+    if (preparedSelect == null) {
+      preparedSelect = getPostgresConnection().prepareStatement(
+          String.format("SELECT k, v FROM %s WHERE region = ? AND k = ?",
+                        getTableName()));
+    }
+    return preparedSelect;
+  }
+
+  @Override
+  public long doRead() {
+    Key key = getSimpleLoadGenerator().getKeyToRead();
+    if (key == null) {
+      // There are no keys to read yet.
+      return 0;
+    }
+
+    try {
+      PreparedStatement statement = getPreparedSelect();
+      statement.setString(1, getRegion(key.asNumber()));
+      statement.setString(2, key.asString());
+      try (ResultSet rs = statement.executeQuery()) {
+        if (!rs.next()) {
+          LOG.error("Read key: " + key.asString() + " expected 1 row in result, got 0");
+          return 0;
+        }
+
+        if (!key.asString().equals(rs.getString("k"))) {
+          LOG.error("Read key: " + key.asString() + ", got " + rs.getString("k"));
+        }
+        LOG.debug("Read key: " + key.toString());
+
+        key.verify(rs.getString("v"));
+
+        if (rs.next()) {
+          LOG.error("Read key: " + key.asString() + " expected 1 row in result, got more");
+          return 0;
+        }
+      }
+    } catch (Exception e) {
+      LOG.fatal("Failed reading value: " + key.getValueStr(), e);
+      return 0;
+    }
+    return 1;
+  }
+
+  private PreparedStatement getPreparedInsert() throws Exception {
+    if (preparedInsert == null) {
+      preparedInsert = getPostgresConnection().prepareStatement(
+          String.format("INSERT INTO %s (region, k, v) VALUES (?, ?, ?)",
+                        getTableName()));
+    }
+    return preparedInsert;
+  }
+
+  @Override
+  public long doWrite(int threadIdx) {
+    Key key = getSimpleLoadGenerator().getKeyToWrite();
+    if (key == null) {
+      return 0;
+    }
+
+    int result = 0;
+    try {
+      PreparedStatement statement = getPreparedInsert();
+      // Prefix hashcode to ensure generated keys are random and not sequential.
+      statement.setString(1, getRegion(key.asNumber()));
+      statement.setString(2, key.asString());
+      statement.setString(3, key.getValueStr());
+      result = statement.executeUpdate();
```

Comment:
Can we ensure that this returns a 1 as well?

Reply:
Good point, but we don't do it for any of the sample apps, so would prefer to keep this consistent with the others for now. We can do that change as a general refactoring.